### PR TITLE
Fix Android/iOS compile for CURL

### PR DESCRIPTION
**Pull request by xsacha**
There are a few tests that the CMakeLists attempts to run on the host system. By setting the variables to what the result would be, it avoids this attempt and successfully compiles.
Tested on Android.

### CODE REVIEW

File: docs/rtfd-css [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit ea8d83e77ef1c07ebd31f2e6dbf7c081f55ce06b
```
Comment (ruslo) [inline docs/rtfd-css]: Not related I guess.

File: cmake/projects/CURL/hunter.cmake [hunk 2]
```diff
@@ -27,15 +27,30 @@ hunter_add_version(
...
+            HAVE_GLIBC_STRERROR_R__TRYRUN_OUTPUT=""
+            HAVE_POSIX_STRERROR_R=0
+            HAVE_POSIX_STRERROR_R__TRYRUN_OUTPUT=""
+            HAVE_POLL_FINE_EXITCODE=0
+     )
+else()
+    hunter_cmake_args(
+        CURL
+        CMAKE_ARGS
+            BUILD_CURL_TESTS=OFF
```
Comment (ruslo) [inline cmake/projects/CURL/hunter.cmake]: This part is common for both configuration. Use separate variable like `_curl_cmake_args`:

```cmake
set(_curl_cmake_args BUILD_CURL_TESTS=OFF BUILD_CURL_EXE=OFF ...)
if(ANDROID OR IOS)
  list(APPEND _curl_cmake_args HAVE_GLIBC_STRERROR_R=1 HAVE_GLIBC_STRERROR_R__TRYRUN_OUTPUT="" ...)
endif()
```

**Comment by xsacha** [comment]
Thanks. Updated.

File: cmake/projects/CURL/hunter.cmake [hunk 3]
```diff
@@ -27,6 +27,15 @@ hunter_add_version(
 )
 
 
+if (ANDROID OR IOS)
+  set(_curl_cmake_args
```
Comment (ruslo) [inline cmake/projects/CURL/hunter.cmake]: If somebody set this variable outside and platform is not Android/iOS we can carry junk to the option. Just add `set(_curl_cmake_args "")` for `else` condition.

Comment (xsacha) [inline cmake/projects/CURL/hunter.cmake]: Fixed that. Thanks.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1086
* https://travis-ci.org/ingenue/hunter/builds/169129812

Note that there is no iOS/Android toolchain in matrix, we do test only the fact that change doesn't broke old code. After this pull request will be merged feel free to send pull request to [pkg.curl](https://github.com/ingenue/hunter/tree/pkg.curl) branch with uncommented iOS/Android entries.

**Comment by ruslo** [comment]
Will be released with https://github.com/ruslo/hunter/pull/552

**Pull request merged**